mlflow/tracking/_model_registry/client.py

```diff
@@ -207,18 +211,30 @@ def create_model_version(
             # local_model_path since old model registry store implementations may not
             # support the local_model_path argument.
             mv = self.store.create_model_version(name, source, run_id, tags, run_link, description)
-        if await_creation_for and await_creation_for > 0:
+        if (
+            not isinstance(self.store, UcModelRegistryStore)
```

Comment:
Adding this clause because it doesn't make sense to await_creation_for UC model versions.

Reply:
@jerrylian-db @BenWilson2 @harupy what do y'all think of adding a new method (`def await_model_version_creation(model_version, await_creation_for)`)  to the model registry abstract store to help customize this behavior at the store level, instead of having store-specific logic at this caller layer? Then we could:
* Have a default implementation that does what the existing logic (before this PR) does, i.e. waits till the model version becomes READY
* Override the implementation in [DatabricksWorkspaceModelRegistryRestStore](https://github.com/mlflow/mlflow/blob/master/mlflow/store/model_registry/databricks_workspace_model_registry_rest_store.py#L17) to additionally include the hint to use UC
* (optional) Override the implementation in `UcModelRegistryStore)` to be a no-op

Reply:
> Override the implementation in DatabricksWorkspaceModelRegistryRestStore to additionally include the hint to use UC

Minor: to enable this we might need to put the abstract class implementation into a common util where we can customize / pass in the timeout hint

Reply:
@smurching That makes a lot of sense. Were you thinking of exposing this timeout override via multiple vectors? (environment variable & function arguments to the call to register the model)